### PR TITLE
Fix #475

### LINKED ISSUE #475: syntax: end delimiter of a region highlighted incorrectly
When a syntax region is declared with the `transparent` attribute and a `matchgroup` equal to the region’s group, the `end` delimiter of the region is transparent, but shouldn’t.

Here’s a reproduction:

```
vim -u NONE -i NONE -N +'hi R ctermbg=Red' +'syn region R matchgroup=R transparent start=/b/ end=/d/' +'call setline(1, "abcde")'
```

Only `b` is highlighted in red; `d` should be as well.

### PR DESCRIPTION
**Pull request by dcondrey**
Fix #475

### CODE REVIEW

File: src/syntax.c [hunk 1]
```diff
@@ -2584,6 +2584,12 @@ update_si_attr(int idx)
 		sip->si_flags |= HL_TRANS_CONT;
 		sip->si_cont_list = CUR_STATE(idx - 1).si_cont_list;
 	    }
+		if (spp->sp_type == ST_REG && spp->sp_end_pattern != NULL &&
+			sip->si_attr ==
+				syn_get_match_attr(spp->sp_end_pattern, 'matchgroup'))
```
Comment (yegappan) [inline src/syntax.c]: Can you add a test for this?

File: src/syntax.c [hunk 2]
```diff
@@ -2584,6 +2584,12 @@ update_si_attr(int idx)
...
+	    {
+		sip->si_attr = CUR_STATE(idx - 1).si_attr;
+	    }
```
Comment (yegappan) [inline src/syntax.c]: Coding style issue:  For a single statement in a `if`, curly braces are not used.

**Review by yegappan — CHANGES_REQUESTED** [review]

**Comment by mattn** [comment]
Compilation errors.
```
syntax.c: In function ‘update_si_attr’:
syntax.c:2587:37: error: ‘ST_REG’ undeclared (first use in this function); did you mean ‘DT_REG’?
 2587 |                 if (spp->sp_type == ST_REG && spp->sp_end_pattern != NULL &&
      |                                     ^~~~~~
      |                                     DT_REG
syntax.c:2587:37: note: each undeclared identifier is reported only once for each function it appears in
syntax.c:2587:52: error: ‘synpat_T’ {aka ‘struct syn_pattern’} has no member named ‘sp_end_pattern’; did you mean ‘sp_pattern’?
 2587 |                 if (spp->sp_type == ST_REG && spp->sp_end_pattern != NULL &&
      |                                                    ^~~~~~~~~~~~~~
      |                                                    sp_pattern
syntax.c:2589:33: warning: implicit declaration of function ‘syn_get_match_attr’; did you mean ‘syn_get_stack_item’? [-Wimplicit-function-declaration]
 2589 |                                 syn_get_match_attr(spp->sp_end_pattern, 'matchgroup'))
      |                                 ^~~~~~~~~~~~~~~~~~
      |                                 syn_get_stack_item
syntax.c:2589:57: error: ‘synpat_T’ {aka ‘struct syn_pattern’} has no member named ‘sp_end_pattern’; did you mean ‘sp_pattern’?
 2589 |                                 syn_get_match_attr(spp->sp_end_pattern, 'matchgroup'))
      |                                                         ^~~~~~~~~~~~~~
      |                                                         sp_pattern
syntax.c:2589:73: warning: character constant too long for its type
 2589 |                                 syn_get_match_attr(spp->sp_end_pattern, 'matchgroup'))
      |                                                                         ^~~~~~~~~~~~
make: *** [Makefile:3459: objects/syntax.o] エラー 1
```

**Comment by mattn** [comment]
@dcondrey please do not send patch without compilation checks. At least, single quotes can not be used for C languages. Is this generated by ChatGPT? If you used ChatGPT, we can't merge this since the patch does not have a copyright.

**Pull request closed without merging**